### PR TITLE
build: bump revm to 7.1

### PR DESCRIPTION
**Pull request by Wodann**
**To Do**
- [x] Change `git` dependency once https://github.com/bluealloy/revm/pull/1163 has been merged
- [x] Fix clippy errors and do cleanup
- [x] Change `git` dependency once https://github.com/bluealloy/revm/pull/1166 has been merged

### CODE REVIEW

File: crates/edr_eth/Cargo.toml [hunk 1]
```diff
@@ -18,8 +18,7 @@ reqwest = { version = "0.11", features = ["blocking", "json"] }
 reqwest-middleware = { version = "0.2.4", default-features = false }
 reqwest-retry = { version = "0.3.0", default-features = false }
 reqwest-tracing = { version = "0.4.7", default-features = false, optional = true }
-revm-primitives = { git = "https://github.com/Wodann/revm", rev = "d6fea43", version = "1.3", default-features = false }
-# revm-primitives = { path = "../../../revm/crates/primitives", version = "1.3", default-features = false }
+revm-primitives = { git = "https://github.com/bluealloy/revm", rev = "f36f90e", version = "2.1", default-features = false }
```
Comment (agostbiro) [inline crates/edr_eth/Cargo.toml:21]: Is the `2.1` version correct?

File: crates/edr_evm/src/state/overrides.rs [hunk 2]
```diff
@@ -247,34 +249,100 @@ impl TryFrom<StateOverrideOptions> for StateOverrides {
...
     }
 
     fn storage(&self, address: Address, index: U256) -> Result<U256, Self::Error> {
         self.overrides
-            .account_storage_at(self.state, &address, &index)
+            .account_storage_at(&self.state, &address, &index)
     }
 }
+
+// // TODO: Remove this once trait upcasting is supported
```
Comment (agostbiro) [inline crates/edr_evm/src/state/overrides.rs:284]: Can this and the lines below be removed?

File: crates/edr_napi/index.d.ts [hunk 3]
```diff
@@ -313,7 +313,7 @@ export const enum ExceptionalHalt {
   /** Error on created contract that begins with EF */
   CreateContractStartingWithEF = 12,
   /** EIP-3860: Limit and meter initcode. Initcode size limit exceeded. */
-  CreateInitcodeSizeLimit = 13
+  CreateInitCodeSizeLimit = 13
```
Comment (agostbiro) [inline crates/edr_napi/index.d.ts:316]: Why does git think this a change? I can't see any 😅

File: crates/edr_solidity/Cargo.toml [hunk 4]
```diff
@@ -5,4 +5,4 @@ edition = "2021"
 
 [dependencies]
 edr_eth = { version = "0.2.0-dev", path = "../edr_eth" }
-revm = { git = "https://github.com/Wodann/revm", rev = "93d8ebe", version = "3.3", default-features = false, features = ["dev", "serde", "std"] }
+revm = { version = "6.1", default-features = false, features = ["dev", "serde", "std"] }
```
Comment (agostbiro) [inline crates/edr_solidity/Cargo.toml:8]: Should we use the same `revm` version in all crates? I noticed here it's the released `6.1` while earlier it was still unreleased `6.2-dev`

**Review by agostbiro — COMMENTED** [review]
Thanks this is huge work @Wodann 🙏 Looks good overall at first pass, but I'll need a second for sure

File: crates/edr_napi/index.d.ts (hunk 3 above)
Comment (agostbiro) [reply crates/edr_napi/index.d.ts:316]: Ah nvm with vertical layout it's obvious it's a capitalization change

File: crates/edr_eth/Cargo.toml (hunk 1 above)
Comment (Wodann) [reply crates/edr_eth/Cargo.toml:21]: See https://github.com/NomicFoundation/hardhat/pull/4968#discussion_r1516587639

File: crates/edr_evm/src/state/overrides.rs (hunk 2 above)
Comment (Wodann) [reply crates/edr_evm/src/state/overrides.rs]: Addressed by https://github.com/NomicFoundation/hardhat/pull/4968/commits/12a22bf9d1642a3bf8e42dfcb18f94b522fd32d7

**Pull request merged**